s3keyring/s3.py

```diff
@@ -297,6 +300,49 @@ def delete_password(self, service, username):
             print("WARNING: {}/{} not found in OS keyring".format(
                 service, username))
 
+    def build_cache(self):
+        """Builds the cache file for the namespace"""
+        cache = {}
+
+        # Get all objects in the namespace
+        objects = list(self.bucket.objects.filter(Prefix=self.namespace))
+
+        for obj in objects:
+            if not obj.key.endswith('secret.b64'):
+                # Do not cache any files that aren't secrets
+                continue
+            # Get the service and username
+            key_data = obj.key.split('/')
+            service = key_data[1]
+            username = key_data[2]
+
+            # Fetch the secret
+            password = self.get_password(service, username)
+
+            # Add to the cache
+            cache[service] = {username: password}
+
+        # write the cache file to S3
+        key = "{}/cache.json".format(self.namespace)
```

Comment:
Can we name this something more purpose-obvious for s3 browsers? Might also cut down on potential clashes.